spec/solver_spec.rb

```diff
@@ -0,0 +1,53 @@
+require_relative('../solver')
+
+describe Solver do
+  before(:all) do
+    @solver = Solver.new
+  end
+
+  context 'test the factorial method' do
+    it 'factorial 4 should return 24' do
+      num = 4
+      result = @solver.factorial(num)
+      expect(result).to eq(24)
+    end
+  end
+
```

Comment:
- [ ] I would recommend adding more tests when `num = 0` the expected result is `1` or when the value is negative, it would raise an exception.

 Currently, I tried these tests but the tests are failing. Please update the method as well accordingly.